modules/u-root

```diff
@@ -7,20 +7,32 @@
 
 u-root_url := github.com/u-root/u-root
 u-root_output := $(build)/$(BOARD)/u-root.cpio
+UROOT_CMDS ?=
+ifeq ($(shell test -e build/$(BOARD)/u-root.cpio && echo -n yes),yes)
+$(shell rm $(build)/$(BOARD)/u-root.cpio)
+endif
 
 export GOPATH=$(build)/go
+src_cmds := $(foreach cmd,$(UROOT_CMDS),github.com/u-root/u-root/cmds/$(cmd))
 
 $(GOPATH):
 	$(call do,GO GET,$(u-root_url),\
 		go get $(u-root_url) \
 	)
 
 $(u-root_output): $(GOPATH)
```

Comment:
The rm shouldn't be required -- I think the problem is that the dep is on the GOPATH directory, which will have been created the first time, rather than a dep on something intermediate.

Reply:
Should the ifeq tests be done in the shell rather than the Makefile? I think you're trying to allow the board to provide its own set of u-root commands; perhaps rather than specializing just this command it would make sense to have a boards/$(BOARD)/u-root subdirectory that is copied into the gopath directory.

(Also, this doesn't make it possible to do multiple boards that use different files without potentially forcing a clean rebuild)

Reply:
I will have a look to the rm stuff tomorrow, you are right about it I believe. I was so focused at recompiling my u-root stuff each time I kicked it off, that I implemented it using brute force

Reply:
the UROOT_CMDS is just a list of commands which will be integrated into the u-root ramfs, something like "pxeboot boot cat mount rsdp" reducing the initramfs to the smallest size and with only the required commands.

Reply:
Just made an update with a "clean" rule definition into the u-root.config module